src/feature/dirauth/process_descs.c

```diff
@@ -381,28 +374,27 @@ dirserv_would_reject_router(const routerstatus_t *rs)
 {
   uint32_t res;
 
-  res = dirserv_get_status_impl(rs->identity_digest, rs->nickname,
-                                rs->addr, rs->or_port,
-                                NULL, NULL, LOG_DEBUG);
+  res = dirserv_get_status_impl(rs->identity_digest,
+                                rs->ed25519_signing_key.pubkey, rs->nickname,
```

Comment:
What if the router does not have an ed25519 key?
This code will crash.

Reply:
0d36d1f514e308d741f2ba232276115458aa86f8